sigopt/xgboost/defaults.py

```diff
@@ -0,0 +1,17 @@
+SEARCH_BOUNDS = [
+  {'name': 'num_boost_round',   'type': 'int',                              'bounds': {'min': 1, 'max': 2}},
+  {'name': 'eta',               'type': 'double', 'transformation': 'log',  'bounds': {'min': 0.001, 'max': 1}},
+  {'name': 'gamma',             'type': 'double',                           'bounds': {'min': 0, 'max': 5}},
+  {'name': 'max_depth',         'type': 'int',                              'bounds': {'min': 1, 'max': 16}},
+  {'name': 'lambda',            'type': 'double',                           'bounds': {'min': 0, 'max': 10}},
+  {'name': 'alpha',             'type': 'double',                           'bounds': {'min': 0, 'max': 10}},
+  {'name': 'min_child_weight',  'type': 'double',                           'bounds': {'min': 0, 'max': 10}},
+  {'name': 'max_delta_step',    'type': 'double', 'transformation': 'log',  'bounds': {'min': 0.001, 'max': 10}}
+]
```

Comment:
can add a few of the categorical ones such as `growth policy` or `booster`.

Reply:
I'm against picking a search space for the categorical right now because there are conditional dependencies on many of them that I don't quite understand (e.g., grow_policy depends on tree_method). If the user wants to explicitly force the experiment to try different ones out, but otherwise I'm not comfortable setting them myself

Reply:
ok. what about automatically supplying parameter type for them?

Reply:
Yea that's fine. I will issue a separate PR for this (bound / type checking and supplying types if needed) if that's okay